core/tests/Test.Microsoft.Identity.LabInfrastructure/KeyVaultSecretsProvider.cs

```diff
@@ -110,14 +150,41 @@ private async Task<string> AuthenticationCallbackAsync(string authority, string
                     }
                     authResult = await authContext.AcquireTokenAsync(resource, _assertionCert);
                     break;
+                case KeyVaultAuthenticationType.ClientSecret:
+                    ClientCredential cred = new ClientCredential(_config.ClientId, _config.KeyVaultSecret);
+                    authResult = await authContext.AcquireTokenAsync(resource, cred);
+                    break;
                 case KeyVaultAuthenticationType.UserCredential:
                     authResult = await authContext.AcquireTokenAsync(resource, _config.ClientId, new UserCredential());
                     break;
                 default:
                     throw new ArgumentOutOfRangeException();
             }
 
+            _authResult = authResult;
             return authResult?.AccessToken;
         }
+
+        public void Dispose()
+        {
+            Dispose(true);
+            GC.SuppressFinalize(this);
+        }
+
+        protected virtual void Dispose(bool disposing)
+        {
+            if (disposed)
+                return;
+
+            if (disposing)
+            {
+                //No managed resources to dispose.
+            }
+
+            // Disposing unmanaged resources.
+            _keyVaultClient.Dispose();
```

Comment:
>_keyVaultClient.Dispose(); [](start = 12, length = 26)

recommended pattern is to check if _keyVaultClient is null, if not call dispose and set to null... that ensures dispose can be called multiple times